### PR TITLE
[C#] refactor: use Azure.AI.ContentSafety in the moderator

### PR DESCRIPTION
**Pull request by kuojianlu**
## Linked issues

Closes: #475 

#### Change details

- leverage Azure.AI.ContentSafety
- remove api version in moderator options to sync with AzureOpenAIPlanner

**code snippets**:

**screenshots**:

## Attestation Checklist

- [x] My code follows the style guidelines of this project

- I have checked for/fixed spelling, linting, and other errors
- I have commented my code for clarity
- I have made corresponding changes to the documentation (we use [TypeDoc](https://typedoc.org/) to document our code)
- My changes generate no new warnings
- I have added tests that validates my changes, and provides sufficient test coverage. I have tested with:
  - Local testing
  - E2E testing in Teams
- New and existing unit tests pass locally with my changes

### Additional information

> Feel free to add other relevant information below

### CODE REVIEW

File: dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI/AI/Moderator/AzureContentSafetyModerator.cs [hunk 1]
```diff
@@ -15,24 +15,16 @@ namespace Microsoft.TeamsAI.AI.Moderator
...
+        private readonly ContentSafetyClient _client;
 
         /// <summary>
         /// Constructs an instance of the moderator.
         /// </summary>
         /// <param name="options">Options to configure the moderator.</param>
-        /// <param name="logger">A logger instance.</param>
-        /// <param name="httpClient">HTTP client.</param>
-        public AzureContentSafetyModerator(AzureContentSafetyModeratorOptions options, ILogger? logger = null, HttpClient? httpClient = null)
+        public AzureContentSafetyModerator(AzureContentSafetyModeratorOptions options)
```
Comment (swatDong) [inline dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI/AI/Moderator/AzureContentSafetyModerator.cs:24]: Any sample change required?

File: dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI/AI/Moderator/AzureContentSafetyModerator.cs [hunk 2]
```diff
@@ -86,41 +78,57 @@ public async Task<Plan> ReviewPlan(ITurnContext turnContext, TState turnState, P
...
-            }
+                    };
+                }
 
-            return null;
+                return null;
+            }
+            catch (RequestFailedException)
+            {
+                throw;
```
Comment (swatDong) [inline dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI/AI/Moderator/AzureContentSafetyModerator.cs:127]: Here seems just catch and throw, then how about removing the try-catch and let exception to directly throw?

File: dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI/AI/Moderator/AzureContentSafetyModerator.cs (hunk 1 above)
Comment (kuojianlu) [reply dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI/AI/Moderator/AzureContentSafetyModerator.cs:24]: Good reminder, I forgot that. Updated three samples that use the moderator.

File: dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI/AI/Moderator/AzureContentSafetyModerator.cs (hunk 2 above)
Comment (kuojianlu) [reply dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI/AI/Moderator/AzureContentSafetyModerator.cs]: Nice idea, removed try-catch.

File: dotnet/samples/04.ai.b.messageExtensions.gptME/Program.cs [hunk 3]
```diff
@@ -112,10 +112,7 @@
         loggerFactory.CreateLogger<AzureOpenAIPlanner<TurnState>>());
 
     // Create AzureContentSafetyModerator
-    IModerator<TurnState> moderator = new AzureContentSafetyModerator<TurnState>(
-        sp.GetService<AzureContentSafetyModeratorOptions>()!,
-        loggerFactory.CreateLogger<AzureContentSafetyModerator<TurnState>>(),
-        moderatorHttpClient);
+    IModerator<TurnState> moderator = new AzureContentSafetyModerator<TurnState>(sp.GetService<AzureContentSafetyModeratorOptions>()!);
```
Comment (swatDong) [inline dotnet/samples/04.ai.b.messageExtensions.gptME/Program.cs:115]: Remove `moderatorHttpClient` as well?

Comment (kuojianlu) [reply dotnet/samples/04.ai.b.messageExtensions.gptME/Program.cs:112]: Removed, thanks.

**Review by swatDong — APPROVED** [review]

**Review by singhk97 — APPROVED** [review]

**Review by singhk97 — APPROVED** [review]

File: dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI/AI/Moderator/AzureContentSafetyModerator.cs [hunk 4]
```diff
@@ -15,24 +15,16 @@ namespace Microsoft.TeamsAI.AI.Moderator
     public class AzureContentSafetyModerator<TState> : IModerator<TState> where TState : ITurnState<StateBase, StateBase, TempState>
```
[the remark this answers is not in the document]
Comment (kuojianlu) [reply dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI/AI/Moderator/AzureContentSafetyModerator.cs:15]: Good catch. Updated (that of OpenAIModerator as well).

**Pull request merged**